src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java

```diff
@@ -136,31 +90,23 @@ public void apply(Project target) {
 			Configuration compileModsMappedConfig = project.getConfigurations().maybeCreate(entry.getRemappedConfiguration());
 			compileModsMappedConfig.setTransitive(false); // Don't get transitive deps of already remapped mods
 
-			extendsFrom(entry.getTargetConfiguration(project.getConfigurations()), entry.getRemappedConfiguration());
+			extendsFrom(entry.getTargetConfiguration(project.getConfigurations()), entry.getRemappedConfiguration(), project);
 
 			if (entry.isOnModCompileClasspath()) {
-				extendsFrom(Constants.Configurations.MOD_COMPILE_CLASSPATH, entry.getSourceConfiguration());
-				extendsFrom(Constants.Configurations.MOD_COMPILE_CLASSPATH_MAPPED, entry.getRemappedConfiguration());
+				extendsFrom(Constants.Configurations.MOD_COMPILE_CLASSPATH, entry.getSourceConfiguration(), project);
+				extendsFrom(Constants.Configurations.MOD_COMPILE_CLASSPATH_MAPPED, entry.getRemappedConfiguration(), project);
 			}
 		}
 
-		extendsFrom("compileClasspath", Constants.Configurations.MINECRAFT_NAMED);
-		extendsFrom("runtimeClasspath", Constants.Configurations.MINECRAFT_NAMED);
-		extendsFrom("testCompileClasspath", Constants.Configurations.MINECRAFT_NAMED);
-		extendsFrom("testRuntimeClasspath", Constants.Configurations.MINECRAFT_NAMED);
+		extendsFrom("compileClasspath", Constants.Configurations.MINECRAFT_NAMED, project);
```

Comment:
I might advise using the big constants in `JavaPlugin` for all the magic string constants here?